### PR TITLE
adding ssh secret to pod template

### PR DESCRIPTION
**Pull request by mchengat**
## Context

Add sshca secret to pod template

## Objective

Add sshca secret to pod template

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: config/pod.yaml.hbs [hunk 1]
```diff
@@ -128,6 +128,16 @@ spec:
     - mountPath: /opt/sdeventcache
       name: sd-event-cache
     {{/if}}
+  env:
+    {{#if secret_entity}}
```
Comment (adong) [inline config/pod.yaml.hbs:132]: Is the intention here only mounts the `volume` if it has `secrets`?

File: config/pod.yaml.hbs [hunk 2]
```diff
@@ -128,6 +128,16 @@ spec:
...
+      {{#each volume_mounts}}
+      - name: {{this.env_name}}
```
Comment (pritamstyz4ever) [inline config/pod.yaml.hbs:134]: can we add `readOnly: true` to the volume

File: config/pod.yaml.hbs (hunk 1 above)
Comment (pritamstyz4ever) [reply config/pod.yaml.hbs:132]: if this is a flag with `true` or `false` value, please rename it accordingly.

File: config/pod.yaml.hbs (hunk 2 above)
Comment (pritamstyz4ever) [reply config/pod.yaml.hbs:134]: also, why is it looping `volume_mounts`? The keys for `volume_mounts` property are `path`, `type` and `name`

**Review by pritamstyz4ever — CHANGES_REQUESTED** [review]

**Review by pritamstyz4ever — APPROVED** [review]

**Pull request merged**